### ISSUE TITLE
Installation is too complex. I don't mean super complex but still.

### ISSUE THREAD

**Issue by 621625** [strang1ato/nhi#27]
Installation is too complex. I don't mean super complex but still.

I don't want to go through those steps with adding sources and so on. I mean I want to test the package but I will feel more safe if you streamline the installation a bit.

Have you considered adding nhi to debian standard packages?

Restarting the computer is required. That is just too weird. Never heard of a small package that requires the whole computer to be restarted. 

A one-liner that works for both debian and ubuntu would be ideal.

Bonus feature request:

nhistory command mockup:

nhistory is the nhi equivalent to history. Instead of listing inputs it lists output, neatly numbered 1-n. You could also have an equivalent to !! which echoes the last output - would be very useful. Could be a big hit with Linux users. Your list of outputs is too complex, we don't need date and such necessarily. History does it fine and simple.

**Comment by strang1ato** [comment]
Thanks for your the interest in the project. There is already working ppa package for older versions of ubuntu: https://github.com/strang1ato/nhi?tab=readme-ov-file#ubuntu-2110 . However currently nhi requires unstripped versions of bash and zsh to work, and some distros like ubuntu (but not debian) include only the stripped version of shells (dbgsym packages are still stripped from the important data), therefore if I wanted to maintain ppa for ubuntu, I also need  to maintain unstripped ppa for bash and zsh, so at least for now, I plan on adding package for nhi when it will work on stripped shells.

> nhistory command mockup:

That sounds cool! There is already `nhi fetch` which can fetch last n commands (and outputs) from specified shell session, but doing the same for last n commands from all sessions (not only one) in general sounds useful as well.

**Comment by 621625** [comment]
Yes, yes. You should do it. Think about the popularity of history : )

There must be a gazillion people that wish they could conveniently just get a simple thing like the last output instead of the last input. But Linux does not always provide what seems obvious (yet).

**Comment by 621625** [comment]
If you do it and it winds up in the debian main repository, please put my initials bm in the nhistory man page somewhere if possible maybe as a thank you note. That would be cool. nhistory is so useful it would almost be a historical event. I mean 10% or more of all Linux Terminal users might use it on a daily basis eventually as they learn about it I think.

**Comment by strang1ato** [comment]
I mean you can get last n commnads (and outputs) from last session by running `nhi fetch {-1}` see: https://youtu.be/i7F3fJdYXSs?si=Z8ddyyKoM3d4amiI&t=74 .  So your idea is minor but important improvemnt

Thank you for kind words. The nhistory would be implemented as part of `nhi fetch` by ommiting shell session.

To implement this feature, I think, you basically need to scan whole database for n last commands accross all sessions. Little nice algorithm in Golang in cli would the thing (I think)

Also, the word "nhi" stands for "New HIstory", the idea for nhi is to make history but for outputs as well.

> please put my initials bm in the nhistory man page somewhere if possible maybe as a thank you note.

Sure, if you are willing to, you can do it yourself by submitting PR for the feature.

**Comment by 621625** [comment]
Great. If possible please update when it's ready here. I would like to test it. I don't think nhistory is a minor improvement. I think I am a great genius : D

Word-of-the-day:

Basic

**Comment by pexes123** [comment]
Buiding own compatible (unstripped) bash, from transparent sources (ideally the latest patch distributed for a particular distro), is even more complex (see #28 ), but somehow necessary ATM.

PPA solution in its current state unsubscribes from security patches like [this mar24 update for jammy](https://launchpad.net/ubuntu/+source/bash/5.1-6ubuntu1.1). 
Providing different version also breaks the point of LTS' cross testing of frozen versions of essential core libs (in case of bash for example some APT package post-install script)
